packages/database/src/migrations/20210119053017-SomoaCovidCustomisedRawDataDownload-modifies-data.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+import { insertObject, addReportToGroups } from '../utilities/migration';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const dashboardReportCode = 'WS_Covid_Samoa_Country';
+const surveyCode = 'SC1QMIA';
+const newDashboardReport = {
+  id: 'Samoa_Covid_Customised_Raw_Data_Download',
+  dataBuilder: 'surveyDataExport',
+  dataBuilderConfig: {
+    surveys: [
+      {
+        code: surveyCode,
+        name: 'Initial Quarantine Monitoring',
+      },
+    ],
+    exportDataBuilder: {
+      dataBuilder: 'rawDataValues',
+      dataBuilderConfig: {
+        skipHeader: false,
+        surveysConfig: {
+          [surveyCode]: {
+            entityAggregation: {
+              aggregationType: 'REPLACE_ENTITY_ANSWER_WITH_ENTITY_NAME',
+              aggregationConfig: {
+                questionCodes: ['QMIA029'],
```

Comment:
Instead of `questionCodes`, let's use `dataElementCodes` since it refers to the actual object that will be retrieved

Reasons for that:
* Data element and question codes currently match, but this is not a strict requirement in our code (eg a question code could be theoretically different from a data element code)
* A data element is not necessarily associated with a question: it can be linked to a 3rd party api (`weather` service), an `indicator` etc

Reply:
Based on my [comment](https://github.com/beyondessential/tupaia/pull/1944#pullrequestreview-577820540), since this will be used in the `surveyDataExport` builder we probably don't event have to specify which question codes are for an `Entity`. Since this data builder deals with survey export, it should also be able to get information about which survey questions are of `Entity` type, and automatically translate all of them if a `dataBuilderConfig` field is set

Reply:
After a bit more looking around, have found this data builder has a name (`SurveyDataExportBuilder`) which is similar to `survey export` , but it seems like it doesn't deal with the questions type.